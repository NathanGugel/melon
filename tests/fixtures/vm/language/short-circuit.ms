
let a = false && invalid();
let c = true || invalid();

io.print(a);
io.print(c);

let t = true;
let f = false;

io.print(f && invalid() || t);
io.print(f && invalid() && t && (t || invalid()));
io.print(f || (f && invalid()));

let obj = {
    test = f && invalid() || t,
    [f && invalid()] = "correct"
};

io.print(obj.test);
io.print(obj[false]);

if (f && invalid() || t) {
    io.print("if correct");
}

while (f && invalid()) {
    io.print("this should not print");
}

func fn => {
    return f && invalid() || t;
}

io.print(fn());