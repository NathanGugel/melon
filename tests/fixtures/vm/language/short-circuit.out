false
true
true
false
false
true
correct
if correct
true
